go/internal/indexerhandler/launchpad.go

```diff
@@ -0,0 +1,188 @@
+package indexerhandler
+
+import (
+	"encoding/json"
+
+	wasmtypes "github.com/CosmWasm/wasmd/x/wasm/types"
+	"github.com/TERITORI/teritori-dapp/go/internal/indexerdb"
+	"github.com/TERITORI/teritori-dapp/go/pkg/launchpadpb"
+	"github.com/friendsofgo/errors"
+	"go.uber.org/zap"
+)
+
+func (h *Handler) handleExecuteSubmitCollection(e *Message, execMsg *wasmtypes.MsgExecuteContract) error {
+	var jsonData map[string]map[string]interface{}
+	if err := json.Unmarshal(execMsg.Msg.Bytes(), &jsonData); err != nil {
+		return errors.Wrap(err, "failed to unmarshal json")
+	}
+
+	collectionData := jsonData["submit_collection"]["collection"]
+	if collectionData == nil {
+		return errors.New("failed to get collection data")
+	}
+
+	collectionId := e.Events["wasm.collection_id"][0]
+	if collectionId == "" {
+		return errors.New("failed to get collection id")
+	}
+
+	collectionDataJson, err := json.Marshal(collectionData)
+	if err != nil {
+		return errors.Wrap(err, "failed to marshal collection data")
+	}
+
+	project := indexerdb.LaunchpadProject{
+		NetworkID:      h.config.Network.ID,
+		ProjectID:      collectionId,
+		CollectionData: collectionDataJson,
+		CreatorID:      h.config.Network.UserID(execMsg.Sender),
+		Status:         launchpadpb.Status_STATUS_INCOMPLETE,
+	}
+	if err := h.db.Create(project).Error; err != nil {
+		return errors.Wrap(err, "failed to create project")
+	}
+
+	h.logger.Info("submited project", zap.Any("symbol", collectionId))
+
+	return nil
+}
+
+func (h *Handler) handleExecuteUpdateMerkleRoot(e *Message, execMsg *wasmtypes.MsgExecuteContract) error {
+	var jsonData map[string]map[string]interface{}
+	if err := json.Unmarshal(execMsg.Msg.Bytes(), &jsonData); err != nil {
+		return errors.Wrap(err, "failed to unmarshal json")
+	}
+
+	collectionId := jsonData["update_merkle_root"]["collection_id"]
+	if collectionId == "" {
+		return errors.New("failed to get collection id")
+	}
+
+	merkleRoot := jsonData["update_merkle_root"]["merkle_root"]
+	if merkleRoot == "" {
+		return errors.New("failed to get merkle root")
+	}
+
+	// Update collection_data
+	if err := h.db.Exec(`

```

Comment:
should not be done through a gorm tx ? 
(same for most of the request with sequential SQL request): https://gorm.io/docs/transactions.html

Reply:
the indexer already wraps everything in a db tx, you just have to focus on the handler implem